src/net.h

```diff
@@ -113,6 +113,14 @@ struct CSerializedNetMsg
     std::string m_type;
 };
 
+enum class ConnectionType {
+    INBOUND, /**< peer initiated connections */
+    OUTBOUND, /**< full relay connections (blocks, addrs, txns) made automatically. Addresses selected from AddrMan. */
+    MANUAL, /**< connections to addresses added via addnode or the connect command line argument */
+    FEELER, /**< short lived connections used to test address validity */
+    BLOCK_RELAY, /**< only relay blocks to these automatic outbound connections. Addresses selected from AddrMan. */
+    ADDR_FETCH, /**< short lived connections used to solicit addrs */
```

Comment:
I wonder if all of these except `INBOUND` should be prefixed with `OUTBOUND_` to indicate that they are different types of outbound connections, and `OUTBOUND` become `OUTBOUND_FULL_RELAY`.

Reply:
> I wonder if all of these except `INBOUND` should be prefixed with `OUTBOUND_` to indicate that they are different types of outbound connections, and `OUTBOUND` become `OUTBOUND_FULL_RELAY`.

See also https://github.com/bitcoin/bitcoin/pull/19405#issuecomment-651273931

Reply:
Indeed! And I would take this one step further: to make the naming consistent and obvious we can distinguish 3 properties of a connection:
1. inbound or outbound
2. automatic or manual
3. by capabilities: full relay, feeler, block relay, solicit addresses

and then name the constants like prop1_prop2_prop3. Something like:

```
INBOUND_AUTO_FULL
OUTBOUND_AUTO_FULL
OUTBOUND_MANUAL_FULL
OUTBOUND_AUTO_FEELER
OUTBOUND_AUTO_BLOCK
OUTBOUND_AUTO_ADDR
```

Reply:
I see how that would add some clarity, but it's also a bigger pain to maintain (if any other dimension of classification is added, all existing constants may need to be renamed).

Perhaps having property test functions (`IsFull()`, `IsOutbound()`, `IsFeeler()`, ...) on `ConnectionType` gets you some of the advantages of more elaborate type names, without the maintenance burden?

Reply:
introduced property test functions
& these behaviors should be captured in the code comments of each type